packages/protocol/migrations_sol/Migration.s.sol

```diff
@@ -1036,46 +1043,70 @@ contract Migration is Script, UsingRegistry, MigrationsConstants {
     }
   }
 
-  function _setConstitution(address governanceAddress, string memory json) public {
-    bool skipSetConstitution = abi.decode(json.parseRaw(".governance.skipSetConstitution"), (bool));
-    IGovernance governance = IGovernance(governanceAddress);
-    string memory constitutionJson = vm.readFile("./governanceConstitution.json");
-    string[] memory contractsKeys = vm.parseJsonKeys(constitutionJson, "$");
+  function _setConstitution(address _governanceAddress, string memory _json) public {
+    bool skipSetConstitution_ = abi.decode(
+      _json.parseRaw(".governance.skipSetConstitution"),
+      (bool)
+    );
+    IGovernance governance_ = IGovernance(_governanceAddress);
+    string memory constitutionJson_ = vm.readFile("./governanceConstitution.json");
+    string[] memory contractsNames_ = vm.parseJsonKeys(constitutionJson_, "$");
+    string memory selectorJson_;
+    string[] memory functionsWithTypes_;
 
-    if (!skipSetConstitution) {
-      for (uint256 i = 0; i < contractsKeys.length; i++) {
-        // TODO need to handle the special case for "proxy"
-        string memory contractName = contractsKeys[i];
+    if (!skipSetConstitution_) {
+      for (uint256 i = 0; i < contractsNames_.length; i++) {
+        string memory contractName_ = contractsNames_[i];
 
-        // TODO make helper function for string comparison
-        if (keccak256(abi.encodePacked(contractName)) == keccak256(abi.encodePacked("proxy"))) {
+        // skip proxy
+        if (contractName_.equals("proxy")) {
```

Comment:
Could you explain why are we skipping proxy here or rather why is it even in governanceConstitution.json in the first place ?

Reply:
Thanks for raising that -> it seems I missed to combine proxy constitutions with other contracts like in original test:
```
const thresholds = { ...constitution.proxy, ...constitution[contractName] }
```